hypothesis-python/src/hypothesis/internal/conjecture/choice.py

```diff
@@ -25,10 +68,12 @@ def _size_to_index(size, *, alphabet_size):
         return 0
     if alphabet_size == 1:
         return size
-    return (alphabet_size**size - 1) // (alphabet_size - 1)
+    v = (alphabet_size**size - 1) // (alphabet_size - 1)
+    # mypy thinks (m: int) // (n: int) -> Any. assert it back to int.
```

Comment:
Looks like that's actually [the `int.__pow__` annotations](https://github.com/python/typeshed/blob/9f28171658b9ca6c32a7cb93fbb99fc92b17858b/stdlib/builtins.pyi#L279-L301) (and with [literals](https://github.com/python/typeshed/blob/9f28171658b9ca6c32a7cb93fbb99fc92b17858b/stdlib/builtins.pyi#L227-L229) not [annotated-types](https://github.com/annotated-types/annotated-types/)); maybe there's a reasonable way to get an assert in?  Otherwise typecheckers seem unlikely to be happy before we get dependent typechecking for integer ranges 😰

Reply:
😭 seems like any assertion would rely on the range bound of 25 and -20, and we're unbounded here, so this may be the best we can do